### DISCUSSION TITLE
Let AutoPairsAutoLineBreak press <CR> before breaking the line

### DISCUSSION THREAD

**Discussion by ptr7F** [LunarWatcher/auto-pairs#89] [Q&A]
I set the option g:AutoPairsAutoLineBreak = ['{'] so that when "{" is pressed it becomes:
(`|` is cursor position)
```
{
    |
}
```
Now there is a problem. For example, if I write an array such as: `int a[] = {1, 2};`
It will become:
```
int a[] = {
     |
}
```
I don't want it to execute automatically. For example, first enter "{", it will become "{|}", and then press <CR> to become:
```
{
    |
 }
```

These are my vimrc options:
```
g:AutoPairsMapBS = 1
g:AutoPairsMultilineBackspace = 1
g:AutoPairsAutoLineBreak = ['{']
```

**Comment by ptr7F** [comment]
Found a conflict with the autoComplete option in lsp of https://github.com/yegappan/lsp

**Comment by LunarWatcher** [comment]
I'm not sure I follow; what exactly is the problem? The option seems to be working as expected if it auto-expands after pressing { - if you want confirmation before expanding, just `<CR>` is mapped to that by default already, with no relation to the auto-expansion options

**Comment by ptr7F** [comment]
I found that this is not a problem with auto-pairs, but a problem with [yegappan/lsp](https://github.com/yegappan/lsp/issues/364). The plugin has some settings that prevent auto-pairs from working properly.
